### PR TITLE
PWN-7169 - swap ui fixes

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7169

## Description of Work

swap ui fixes

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/FeatureException.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import java.math.BigDecimal
+
+open class FeatureException : Exception() {
+
+    data class NotValidTokenA(
+        val notValidAmount: BigDecimal,
+    ) : FeatureException()
+
+    object RoutesNotFound : FeatureException()
+}
```
Comment (NikolaiEmelianov) [inline app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/FeatureException.kt:12]: https://github.com/p2p-org/key-app-android/pull/1284/files
same

File: app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapState.kt [hunk 2]
```diff
@@ -41,4 +41,19 @@ sealed interface SwapState {
...
+        data class FeatureExceptionWrapper(
+            override val previousFeatureState: SwapState,
+            val featureException: FeatureException,
+        ) : SwapException
+
+        data class OtherException(
+            override val previousFeatureState: SwapState,
+            val exception: Throwable,
+        ) : SwapException
+    }
```
Comment (NikolaiEmelianov) [inline app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapState.kt:58]: https://github.com/p2p-org/key-app-android/pull/1284/files
same

File: app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateAction.kt [hunk 3]
```diff
@@ -1,20 +1,17 @@
...
     data class SlippageChanged(val newSlippageValue: Double) : SwapStateAction()
 
-    data class InitialLoadingFinished(
-        val tokenA: Token.Active,
-        val tokenB: Token
-    ) : SwapStateAction()
+    object InitialLoading : SwapStateAction()
 
     object RefreshRoutes : SwapStateAction()
+    object EmptyAmountTokenA : SwapStateAction()
```
Comment (NikolaiEmelianov) [inline app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/SwapStateAction.kt:14]: https://github.com/p2p-org/key-app-android/pull/1284/files
same

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt [hunk 4]
```diff
@@ -17,24 +25,51 @@ import org.p2p.wallet.R
 import org.p2p.wallet.common.mvp.BaseMvpFragment
 import org.p2p.wallet.databinding.FragmentJupiterSwapBinding
 import org.p2p.wallet.swap.ui.jupiter.main.widget.SwapWidgetModel
+import org.p2p.wallet.utils.args
 import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+import java.util.*
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt:31]: Something is broken with your code style. 
Are you sure you are using `Project`?

**Review by 1sakov — APPROVED** [review]

**Pull request merged**